talk/objectorientation/operators.tex

```diff
@@ -86,6 +86,30 @@
   \end{block}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Friend declarations}
+  \begin{block}{Concept}
+    \begin{itemize}
+      \item Functions/classes can be declared \mintinline{cpp}{friend} within a class scope
+      \item They gain access to all private members
+      \item Useful for operators such as \mintinline{cpp}{a + b}
+      \item Don't abuse this! Don't modify an object in a friend!
```

Comment:
> Don't modify an object in a friend!

What is the rational for this? Why shouldn't I grant a different class the ability to modify me?

Reply:
Of course you can. And there's cases where it might make sense. But more often than not, I see the friends being abused as a convenient way out of a proper refactoring. Especially in HEP in older codes.
What I want to say is:
- They are great for operators where you mostly need read access.
- If you use write access, all bets are off. Don't do stupid things / unexpected things / thread-unsafe things / ...

I didn't know how to say this in one sentence.

Reply:
Hmm, I see. Then maybe you can add:
```suggestion
      \item Don't abuse this! Avoid modifying an object in a friend, use the objects member functions instead.
```
Maybe also split the two sentences into two bullet points.

Reply:
They now read:
```diff
+      \item Don't abuse friends to go around a wrongly designed interface
+      \item Avoid unexpected modifications of class state in a friend
```